lib/scale/encode.go

```diff
@@ -329,6 +329,14 @@ func (se *Encoder) encodeArray(t interface{}) (bytesEncoded int, err error) {
 			n, err = se.encodeByteArray([]byte(elem))
 			bytesEncoded += n
 		}
+	case []common.PeerInfo:
```

Comment:
same for here, I guess there isn't encoding support for arrays of structs either?